src/components/checkbox/GroupableCheckboxReducersUtils.ts

```diff
@@ -93,16 +99,73 @@ export const toggleChildCheckbox = (state: IGroupableCheckboxesState, action: IR
 
 export const toggleParentCheckbox = (state: IGroupableCheckboxesState, action: IReduxAction<IGroupableCheckboxActionPayload>) => {
     const isChecked: boolean = !state.parent.checked;
+    const newCheckboxes: ICheckboxState[] = _.map(state.checkboxes, (checkbox: ICheckboxState) => {
+        return {
+            ...checkbox,
+            checked: checkbox.disabled ? checkbox.checked : isChecked,
+        };
+    });
     return {
         ...state,
         parent: {
+            ...state.parent,
             id: action.payload.id,
             checked: isChecked,
         },
+        checkboxes: newCheckboxes,
+        nbChecked: calculateTotalCheckboxesChecked(newCheckboxes),
+    };
+};
+
+export const disabledParentCheckbox = (state: IGroupableCheckboxesState, action: IReduxAction<IGroupableCheckboxActionPayload>) => {
+    const isDisabled: boolean = !state.parent.disabled;
+    return {
+        ...state,
+        parent: {
+            ...state.parent,
+            id: action.payload.id,
+            disabled: isDisabled,
+        },
+    };
+};
+
+export const disabledChildCheckbox = (state: IGroupableCheckboxesState, action: IReduxAction<IGroupableCheckboxActionPayload>) => {
+    let isDisabled: boolean;
+    const checkboxes: ICheckboxState[] = _.map(state.checkboxes, (checkbox: ICheckboxState) => {
+        if (checkbox.id === action.payload.id) {
+            isDisabled = !checkbox.disabled;

```

Comment:
Is the variable useful elsewhere than a few lines below? You could simply set `disabled: !checkbox.disabled` in the return.